### PR TITLE
Clients: added option to list rse usage per account; Fix #1218

### LINKED ISSUE #1218: New option for rucio list-rse-usage to list usage per account
Motivation
----------
rucio list-rse-usage returns the RSE usage from different sources (Rucio, storage, expired). It would be good to have an option to list the usage per account.

### PR DESCRIPTION
**Pull request by hahahannes**
Clients: added option to list rse usage per account; Fix #1218

### CODE REVIEW

File: bin/rucio [hunk 1]
```diff
@@ -1995,6 +2009,7 @@ You can filter by key/value, e.g.::
     list_rse_usage_parser.set_defaults(which='list_rse_usage')
     list_rse_usage_parser.add_argument(dest='rse', action='store', help='Rucio Storage Element (RSE) name.').completer = rse_completer
     list_rse_usage_parser.add_argument('--history', dest='history', default=False, action='store', help='List RSE usage history. [Unimplemented]')
+    list_rse_usage_parser.add_argument('--per-account', dest='per_account', action='store_true', default=False, help='List RSE usage per account')
```
Comment (bari12) [inline bin/rucio]: `--per-account` sounds a bit weird to me. I think `--show-accounts` or something similar might be better.

File: lib/rucio/api/rse.py [hunk 2]
```diff
@@ -243,17 +244,18 @@ def set_rse_usage(rse, source, used, free, issuer):
...
+def get_rse_usage(rse, issuer, source=None, per_account=False):
     """
     get RSE usage information.
 
     :param rse: The RSE name.
     :param issuer: The issuer account.
     :param source: dictionary of attributes by which the results should be filtered
+    :param per_account: Boolean whether usage should be calculated per account or not
 
     :returns: True if successful, otherwise false.
```
Comment (bari12) [inline lib/rucio/api/rse.py]: I know you did not write this line, but please change to what is actually happening.

File: lib/rucio/core/rse.py [hunk 3]
```diff
@@ -485,30 +486,42 @@ def set_rse_usage(rse, source, used, free, session=None):
...
     """
     get rse usage information.
 
     :param rse: The rse name.
     :param source: The information source, e.g. srm.
     :param rse_id:  The RSE id.
     :param session: The database session in use.
+    :param per_account: Boolean whether the usage should be calculated per account or not.
 
     :returns: True if successful, otherwise false.
```
Comment (bari12) [inline lib/rucio/core/rse.py]: Same with this line

File: lib/rucio/api/rse.py (hunk 2 above)
Comment (hahahannes) [reply lib/rucio/api/rse.py]: More like 'Show accounts usages of RSE'? Also shall I replace the parameter name to 'show_accounts'?

Comment (bari12) [reply lib/rucio/api/rse.py]: I mean the `:returns: True if successful, otherwise false.` should actually be a `:returns: List of RSE usage data`

**Comment by bari12** [comment]
This will need a rebase, infact, it might be better to just re-submit this as an actual patch so things are in order.

**Comment by hahahannes** [comment]
Okay, will do

**Pull request closed without merging**